src/Controller/Component/AuthComponent.php

```diff
@@ -147,7 +154,8 @@ class AuthComponent extends Component
         'logoutRedirect' => null,
         'authError' => null,
         'unauthorizedRedirect' => true,
-        'storage' => 'Session'
+        'storage' => 'Session',
+        'earlyAuth' => false
```

Comment:
Would it be possible to pass an event string that it registers? That way it could be almost anything.